src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryRegEx.cs

```diff
@@ -0,0 +1,274 @@
+                Group g = m.Groups[i];
+
+                if (!int.TryParse(groupName, out _))
+                {
+                    if (!fields.ContainsKey(validName))
+                    {
+                        fields.Add(validName, new NamedValue(validName, StringValue.New(g.Value)));
+                    }
+                    else
+                    {
+                        fields[validName] = new NamedValue(validName, StringValue.New(g.Value));
+                    }
+                }
+
+                if (i > 0)
+                {
+                    subMatches.Add(g.Value);
+                }
+            }
+
+            if (!fields.ContainsKey(SUBMATCHES))
+            {
+                fields.Add(SUBMATCHES, new NamedValue(SUBMATCHES, TableValue.NewSingleColumnTable(subMatches.Select(s => StringValue.New(s)).ToArray())));
+            }
+
+            return RecordValue.NewRecordFromFields(fields.Values);
+        }
+
+        private static DType GetRecordTypeFromRegularExpression(string regularExpression)
+        {
+            Dictionary<string, TypedName> propertyNames = new ();
+            Regex rex = new Regex(regularExpression);
+
+            propertyNames.Add(FULLMATCH, new TypedName(DType.String, new DName(FULLMATCH)));
+            propertyNames.Add(STARTMATCH, new TypedName(DType.Number, new DName(STARTMATCH)));
+            propertyNames.Add(SUBMATCHES, new TypedName(DType.CreateTable(new TypedName(DType.String, new DName(TexlFunction.ColumnName_ValueStr))), new DName(SUBMATCHES)));
+
+            foreach (string groupName in rex.GetGroupNames())
+            {
+                if (!int.TryParse(groupName, out _))
+                {
+                    DName validName = DName.MakeValid(groupName, out _);
+
+                    if (!propertyNames.ContainsKey(validName.Value))
+                    {
+                        propertyNames.Add(validName.Value, new TypedName(DType.String, validName));
+                    }
+                }
+            }
+
+            return DType.CreateRecord(propertyNames.Values);
+        }
+
+        private static FormulaValue CommonMatchImpl(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args, string defaultMatchOptions, Func<string, string, RegexOptions, FormulaValue> impl)
+        {
+            if (args[0] is not StringValue sv0)
+            {
+                return CommonErrors.GenericInvalidArgument(args[0].IRContext);
+            }
+
+            if (args[1] is not StringValue sv1)
+            {
+                return CommonErrors.GenericInvalidArgument(args[1].IRContext);
+            }
+
+            string inputString = sv0.Value;
+            string regularExpression = sv1.Value;
+            string matchOptions = args.Length == 3 ? ((StringValue)args[2]).Value : defaultMatchOptions;
+
+            RegexOptions regOptions = RegexOptions.None;
```

Comment:
>RegexOptions.None [](http://example.com/codeflow?start=38&length=17)

RegexOptions.CultureInvariant? 
Else the regex can read Thread.CurrentCulture on ignore case. (you can verify this with the DoNotUserCulture tests)

Reply:
Good catch.
Was wondering if we shouldn't add Compiled too as we have a cache